### ISSUE TITLE
[M3] MLX train_step + wire training loop

### ISSUE THREAD

**Issue by travisgalloway** [travisgalloway/monica#11]
Part of #2.

**Goal:** Provide the backend training primitive and finish the loop orchestration.

**Acceptance criteria**
- Implement an MLX `train_step(model, inputs, targets, lr) -> {loss, grad_norm}` using `nn.value_and_grad` + optimizer, with gradient accumulation, gradient clipping, and mixed precision (loss scaling if fp16, see #3).
- Finish the `TODO[mac]` checkpoint/logging hooks in `src/train/loop.py`.
- Logging from step 1: loss, val loss/perplexity, LR, grad norm, tokens/sec (W&B or similar).
- Toy ~50 steps: train loss and val perplexity both improve.

**Files:** `src/train/loop.py` (skeleton + `TrainStepFn` seam exist), `src/eval/val_loss.py` (done). Depends on #8.

**Comment by travisgalloway** [comment]
Done — verified on Apple Silicon.

`src/model/mlx_train_step.py` (grad via `nn.value_and_grad`, loss scaling, grad clip) + `src/train/loop.py` are wired and exercised by the smoke gate: loss drops 8.07 -> 1.54 over 50 steps.

Verified on Apple Silicon (macOS arm64, Python 3.14.3, mlx 0.31.2) at `main` @ c20f646. Full suite green (20/20) and the M4 smoke gate passes end to end:

```
[reference] step0 loss=8.06889  step49 loss=1.53557
[resumed]   resumed at step=25  step49 loss=1.53557
[match] post-resume max|loss diff| over steps 25..49 = 1.192e-07
[eval] val_loss=1.5416  val_perplexity=4.6721
SMOKE TEST PASSED ✅  resume is exact and eval runs.
```
